client/rest-high-level/src/test/java/org/elasticsearch/client/ml/NodeAttributesTests.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.ml;
+
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class NodeAttributesTests extends AbstractXContentTestCase<NodeAttributes> {
+
+    public static NodeAttributes createRandom() {
+        int numberOfAttributes = randomIntBetween(1, 10);
+        Map<String, String> attributes = new HashMap<>(numberOfAttributes);
+        for(int i = 0; i < numberOfAttributes; i++) {
+            String val = randomAlphaOfLength(10);
+            attributes.put("key-"+i, val);
+        }
+        return new NodeAttributes(randomAlphaOfLength(10),
+            randomAlphaOfLength(10),
+            randomAlphaOfLength(10),
+            randomAlphaOfLength(10),
+            attributes);
+    }
+
+    @Override
+    protected NodeAttributes createTestInstance() {
+        return createRandom();
+    }
+
+    @Override
+    protected NodeAttributes doParseInstance(XContentParser parser) throws IOException {
+        return NodeAttributes.PARSER.parse(parser, null);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
this should be `true`

Reply:
No, it shouldn't, because the internal map is just a map of strings and objects. Consequently, when random fields are added to the serialized object, and it is de-serialized, they are no longer equal as the individual internal attribute fields are not well defined.

Reply:
I don't think that is the case. The unknown fields will only be at the first level, not under the `attributes` object.

Reply:
Unknown fields are injected recursively into internal objects. Example from a failing test for when this is `true`
```
*****OBJECT PRE SHUFFLE + RANDOM FIELDS ********
{
  "id" : "IeAYBTguoP",
  "name" : "LOmZFvZJJN",
  "ephemeral_id" : "pNZZYyuHQC",
  "transport_address" : "oxceWLCGyt",
  "attributes" : {
    "key-1" : "EukbVGneBx",
    "key-0" : "CswxpsJUWa",
    "key-5" : "GEaErpPvvc",
    "key-4" : "aQotGStjzI",
    "key-3" : "juQFiTIlSy",
    "key-2" : "ynlRGrbNtc"
  }
}
****ATRIBUTES POST SHUFFLE + RANDOM FIELDS*****
{
 "key-1" : "EukbVGneBx",
 "key-0" : "CswxpsJUWa",
 "key-5" : "GEaErpPvvc",
 "key-4" : "aQotGStjzI",
 "key-3" : "juQFiTIlSy",
 "NrIvvAzJin" : 5240957203306958078,
 "key-2" : "ynlRGrbNtc"
}
```

Reply:
Do they?! Sorry Ben, I was really not expecting they would. Not a big fan of this. OK, you can ignore my comments about unknown fields in test. However, I wonder if we should add a manual test adding a top-level unknown field to safe-guard this as it is critical for the client forward compatibility.

Reply:
@dimitris-athanasiou I was thinking for a separate PR adding a "shallow" check that does not recursively inject unknown fields. That way we can reuse that code for other other objects that have this same behavior.

Reply:
@dimitris-athanasiou figured out how to only create random fields for the toplevel part of the object.

```
    @Override
    protected Predicate<String> getRandomFieldsExcludeFilter() {
        return field -> !field.isEmpty();
    }
```
This predicate means that for anything whose field path is not `""` we should not add random fields. `""` is the same as the root path. 

`XContentTestUtils.insertRandomFields` calls `XContentTestUtils.getInsertPaths` and then utilized that supplied predicate to filter them.